lib/src/clojure_lsp/feature/completion.clj

```diff
@@ -279,27 +271,46 @@
                            (string/starts-with? (:name %) name))))
          (mapv #(element->completion-item % alias :alias-keyword)))))
 
+(defn ^:private var-defs-in-ns [analysis ns]
+  ;; TODO: maybe better to use queries/find-all-var-definitions, or to refactor
+  ;; queries so that they are composable as transducer transformations.
+  (->> analysis
+       (mapcat val)
+       (filter #(= ns (:ns %)))
+       (filter #(= :var-definitions (:bucket %)))))
```

Comment:
Oh, this one is probably a query should be on queries.clj and we missed

Can we use `identical?` here instead of `=`, move this as the first predicate and convert this to use transducers just like we are using on queries.clj? It should increase even more the performance

Reply:
Yeah, using a transducer is a good idea.

Is `identical?` any different from `=` when comparing keywords? 

I think moving this to queries.clj should be part of a bigger discussion though. queries.clj already has `find-var-definitions`, which is similar. But, it can't be reused because it only works on a single file, which isn't applicable in this case. There's also `find-all-var-definitions`, which doesn't let me filter by namespace soon enough. I'm hesitant to introduce a third variation that's only used in one place. 

I think it'd be nice to separate the pieces of queries.clj into several filter functions that can be recombined in transducers. That is, instead of a function like `find-var-definitions`, which says "find analysis for file AND filter to var definitions", create two functions "find analysis for file" and "filter to var definitions". Then they could be recombined in transducers with `comp` as needed. For example, this completion code could first filter by ns, then comp that with the shared "filter to var definitions" query.

I'd rather not do all of that as part of this PR, but it's something to think about. What do you think?

Reply:
> Is identical? any different from = when comparing keywords?

Yes, [it has a better peformance](https://groups.google.com/g/clojure/c/cqYUG4b9uWE)

> I'd rather not do all of that as part of this PR, but it's something to think about. What do you think?

Yes, makes sense, it's not that easy since comparing a ns could be different if a var-definition or a var-usage for example.
I'm ok not changing this on this PR

Reply:
> it has a better peformance

Good to know.

> could be different if a var-definition or a var-usage

Because of aliases? Aliases introduce so much subtlety!

Reply:
not only alias, but there is `:to`, `:from`, `:ns`, multiple ways to check a namespace depending on the kondo analysis you are iterating